algorithms/templates/shortest_ss_any_wt.gtmp

```diff
@@ -1,73 +1,87 @@
-  ##### Initialization: Set the best-known distance to a maxValue (pseudo-infinite)
-        Source = SELECT s FROM Source:s
-                 POST-ACCUM sourceName = s.name;        # Get the name of the source vertex
-        VSET = SELECT s FROM VSET:s
-               POST-ACCUM s.@dist = maxVal,
-                          s.@path = sourceName;
-        Source = SELECT s FROM Source:s               # The distance FROM s to itself is 0.
-                 POST-ACCUM s.@dist = 0,
-                            s.@visited = true;
-  
-  ##### Do N-1 iterations: Consider whether each edge lowers the best-known distance.
-        FOREACH i IN RANGE[1, VSET.size()-1] DO
-                VSET = SELECT s
-                       FROM VSET:s -(*edge-types*:e)-> :t
-                       ACCUM IF s.@visited THEN
-                                IF s.@dist + e.*edge-weight* < t.@dist THEN
-                                        t.@dist = s.@dist + e.*edge-weight*,             # @dist is a MinAccum
-                                        t.@path = s.@path +"-"+ t.name
-                                END,
-                                t.@visited += true
-                             END;
+              
+        ##### Get the connected vertices
+        start = SELECT s
+                FROM start:s
+                ACCUM s.@minPath += pathTuple(0, s),
+                      s.@visited = TRUE,
+                      s.@path += s;
+        WHILE start.size() > 0 DO
+                start = SELECT t
+                        FROM start:s -(*edge-types*:e)-> :t
+                        WHERE NOT t.@visited
+                        ACCUM t.@visited = TRUE;
+                total = total UNION start;
         END;
-             
-  ##### Check for loops with net negative weight #####
-        VSET = SELECT s
-               FROM VSET:s -(*edge-types*:e)-> :t
-               ACCUM @@hasNegLoop +=s.@dist + e.*edge-weight* < t.@dist;
-         
-        IF @@hasNegLoop THEN        
-                RAISE negLoop (errMsg);
+        
+  ##### Do V-1 iterations: Consider whether each edge lowers the best-known distance.
+        iter = total.size() - 1;    # the max iteration is V-1
+        WHILE TRUE LIMIT iter DO 
+                tmp = SELECT s
+                      FROM total:s -(*edge-types*:e)-> :t
+                      ACCUM 
+                            IF s.@minPath.size()>0 AND s.@minPath.top().dist < GSQL_INT_MAX THEN
+                                t.@minPath += pathTuple(s.@minPath.top().dist + e.*edge-weight*, s)
+                            END;     
+        END;
+        
+  ##### Detect negative cycle
+        tmp = SELECT s
+              FROM total:s -(*edge-types*:e)-> :t
+              ACCUM 
+                    IF s.@minPath.top().dist + e.*edge-weight* < t.@minPath.top().dist THEN
+                            negativeCycle = TRUE
+                    END;
+        IF negativeCycle THEN 
+                PRINT "Negative Cycle Detected.";
```

Comment:
There is better way of throwing an error please refer to:

https://docs.tigergraph.com/dev/gsql-ref/querying/exception-statements